chore: cleanup

DanielRivers opened the pull request:
# Explain your changes

Removing redundant dependencies and cleaning up some bits.

# Checklist

- [x] I have read the [“Pull requests” section](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md#pull-requests) in the [contributing guidelines](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I agree to the terms within the [code of conduct](https://github.com/kinde-oss/.github/blob/main/.github/CODE_OF_CONDUCT.md).

🛟 _If you need help, consider asking for advice over in the [Kinde community](https://thekindecommunity.slack.com)._

DanielRivers committed:
chore: release v2.13.1

DanielRivers committed:
2.14.0-1

DanielRivers committed:
2.14.0-2

DanielRivers committed:
2.14.0-3

DanielRivers committed:
2.14.0-4

DanielRivers committed:
2.14.0-5

DanielRivers committed:
chore: clean up some dependancies and lint issues

dtoxvanilla1991 commented:
@copilot can you resolve conflicts, address comments and make sure the build passes? Do not open a new PR, it must be done in this same PR.

dtoxvanilla1991 commented:
I'll take over this PR.

Copilot committed:
chore: merge main into chore/repo-tidy, resolve conflicts

Co-authored-by: dtoxvanilla1991 <73205087+dtoxvanilla1991@users.noreply.github.com>

Copilot committed:
chore: resolve merge conflicts and fix lint/build issues

Agent-Logs-Url: https://github.com/kinde-oss/kinde-typescript-sdk/sessions/bd60ec36-9769-42c7-b781-b4d424dc5f3f

Co-authored-by: dtoxvanilla1991 <73205087+dtoxvanilla1991@users.noreply.github.com>

Copilot committed:
chore: fix typo 'dependancies' -> 'dependencies' in CHANGELOG

Agent-Logs-Url: https://github.com/kinde-oss/kinde-typescript-sdk/sessions/bd60ec36-9769-42c7-b781-b4d424dc5f3f

Co-authored-by: dtoxvanilla1991 <73205087+dtoxvanilla1991@users.noreply.github.com>

dtoxvanilla1991 commented on sdk-version.js:
```diff
@@ -1,17 +1,20 @@
   const fileContent = await fsPromises.readFile(pathToFile, 'utf8');
 
+  // eslint-disable-next-line no-undef
   const isCleanArg = process.argv[2] === 'clean';
```
Should be addressed, please check.

dtoxvanilla1991 commented on lib/sdk/utilities/token-utils.ts:
```diff
@@ -40,10 +40,10 @@ export const commitTokenToSession = async (
       }
-    } catch (e) {
+    } catch (error) {
       throw new KindeSDKError(
         KindeSDKErrorCode.INVALID_TOKEN_MEMORY_COMMIT,
-        `Attempting to commit invalid ${type} token "${token}" to memory`
+        `Attempting to commit invalid ${type} token "${token}" to memory: ${(error as Error).message}`
       );
```
Should be addressed, please check.

dtoxvanilla1991 commented on lib/sdk/clients/types.ts:
```diff
@@ -15,8 +15,8 @@ export interface BrowserPKCEClientOptions extends AuthorizationCodeOptions {
-export interface PKCEClientOptions extends AuthorizationCodeOptions {}
-export interface CCClientOptions extends ClientCredentialsOptions {}
+export type PKCEClientOptions = AuthorizationCodeOptions;
+export type CCClientOptions = ClientCredentialsOptions;
```
Should be addressed, please check.

dtoxvanilla1991 commented on CHANGELOG.md:
```diff
@@ -44,14 +44,43 @@
+- chore: release v2.13.0 [`557836b`](https://github.com/kinde-oss/kinde-typescript-sdk/commit/557836bfa077bed72569babc052f5e73e6356263)
+- Merge pull request #80 from kinde-oss/feat/session-persistence [`057e5c0`](https://github.com/kinde-oss/kinde-typescript-sdk/commit/057e5c0857f104f4f4f9c5f0a5f48deea0af0dce)
+- feat: set sessionManager persistent property depending on accessToken flag [`7fe1a6c`](https://github.com/kinde-oss/kinde-typescript-sdk/commit/7fe1a6c61053d804ec1e688602880ec29ce730cb)
+
```
Should be addressed, please check.

dtoxvanilla1991 commented:
@copilot the build is failing. Please resolve all issues on this.

Copilot committed:
fix: convert sdk-version.js to CommonJS to fix Node 18 build failure

Agent-Logs-Url: https://github.com/kinde-oss/kinde-typescript-sdk/sessions/b407896f-67d9-48c9-9091-e016f10972a6

Co-authored-by: dtoxvanilla1991 <73205087+dtoxvanilla1991@users.noreply.github.com>

dtoxvanilla1991 commented:
@coderabbitai re-review

dtoxvanilla1991 commented:
@copilot 
Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

Outside diff comments:
In `@lib/sdk/oauth2-flows/AuthCodeAbstract.ts`:
- Around line 293-314: The auth URL builder in AuthCodeAbstract.ts currently
merges options.authUrlParams (via rest) into searchParamsObject, allowing
user-controlled "state"; update the merge to explicitly exclude "state" so
framework-generated state is always used (do not allow
options.authUrlParams.state to override), and remove any logic in
AuthorizationCode and AuthCodeWithPKCE that reads providedState from
options.authUrlParams; ensure searchParamsObject is built from rest without
"state" and that state is only set by the library's state generation/assignment
code.

---

Duplicate comments:
In `@lib/sdk/clients/types.ts`:
- Around line 18-19: PKCEClientOptions and CCClientOptions were changed from
interfaces to type aliases which breaks declaration merging for external
consumers; either revert them back to exported interfaces named
PKCEClientOptions and CCClientOptions or, if you must keep type aliases, add a
changelog entry under "Potentially breaking (types only)" noting the conversion
and advising consumers to update declaration-merged augmentations; update
references to the symbols PKCEClientOptions and CCClientOptions accordingly so
the public API and docs reflect the chosen approach.

In `@lib/sdk/utilities/token-utils.ts`:
- Around line 43-48: The error message in the KindeSDKError thrown inside
token-utils.ts leaks the raw token by interpolating token into the string;
update the KindeSDKError creation (the catch block that throws with
KindeSDKErrorCode.INVALID_TOKEN_MEMORY_COMMIT) to remove the raw token from the
message — either omit the token entirely or replace it with a redacted
placeholder like "[REDACTED]" or token length info — and keep the existing
error.message for context; ensure references remain to KindeSDKError and
KindeSDKErrorCode.INVALID_TOKEN_MEMORY_COMMIT so callers can still handle the
error.

Copilot committed:
fix: prevent authUrlParams.state override, revert type aliases to interfaces, redact token from error message

Agent-Logs-Url: https://github.com/kinde-oss/kinde-typescript-sdk/sessions/62294081-6fd7-4964-a7e8-cdd5efb99f19

Co-authored-by: dtoxvanilla1991 <73205087+dtoxvanilla1991@users.noreply.github.com>
